src/main/java/org/aist/aide/finitehorizonservice/domain/models/FiniteType.java

```diff
@@ -0,0 +1,31 @@
+package org.aist.aide.finitehorizonservice.domain.models;
+
+import java.util.Set;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import org.springframework.data.annotation.Id;
+
+public class FiniteType {
+    @Id
+    private String typeId;
+    @NotNull
+    @NotEmpty
+    private Set<String> cases;
```

Comment:
I think "values" would sound better here instead of "cases". Cases make me think of production systems or use cases.